# T1 Carga e inspección

**Autora:** Jia Xi Shan

**Proyecto P1:** Análisis exploratorio del Titanic (numpy y pandas)

# Carga de datos

In [1]:
import pandas as pd
import numpy as np

# Carga del dataset desde la carpeta Data del repositorio.
# Si no se encuentra (por ejemplo al abrir el notebook en Colab), se descarga
# de la misma fuente original, así que los datos son idénticos.
try:
    df = pd.read_csv("../Data/titanic.csv")
except FileNotFoundError:
    df = pd.read_csv("https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv")

df.shape

(891, 15)

# Mi tarea
Voy a inspeccionar el dataset Titanic para conocer su estructura, dimensiones, tipos de datos y estadísticas descriptivas antes de realizar el análisis.

In [2]:
# Mostramos las primeras filas del dataset.
df.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


In [3]:
# Mostramos información sobre las columnas y los tipos de datos.
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    str    
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    str    
 8   class        891 non-null    str    
 9   who          891 non-null    str    
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    str    
 12  embark_town  889 non-null    str    
 13  alive        891 non-null    str    
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), str(7)
memory usage: 92.4 KB


In [4]:
# Mostramos algunas estadísticas básicas de los datos numéricos.
df.describe()

,survived,pclass,age,sibsp,parch,fare
count,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [5]:
# Mostramos el número de filas y columnas.
df.shape

(891, 15)

## Diccionario de datos

| Columna | Descripción |
|---|---|
| survived | Indica si el pasajero sobrevivió. 0 = No, 1 = Sí |
| pclass | Clase del billete. 1 = Primera, 2 = Segunda, 3 = Tercera |
| sex | Sexo del pasajero |
| age | Edad del pasajero |
| sibsp | Número de hermanos/as o cónyuges a bordo |
| parch | Número de padres/madres o hijos/as a bordo |
| fare | Precio del billete |
| embarked | Código del puerto donde embarcó |
| class | Clase del billete en formato texto |
| who | Categoría del pasajero |
| adult_male | Indica si el pasajero es un hombre adulto |
| deck | Cubierta del barco |
| embark_town | Ciudad donde embarcó |
| alive | Indica si el pasajero sobrevivió |
| alone | Indica si viajaba solo |

## Hallazgo
El dataset tiene 891 pasajeros y 15 columnas. La edad tiene 714 valores y la edad media es de unos 29,70 años. También hay datos numéricos y categóricos, como la edad, el sexo, la clase y la supervivencia.